Makefile

```
@@ -5,11 +5,18 @@ export INTEGRATION_TEST_MAX_INSTANCE_AGE?=86400
 export INTEGRATION_TEST_SUBNET_ID?=integration_test_subnet_id
 export INTEGRATION_TEST_INSTANCE_TAG?=integration_test_instance_tag
 export JOB_ID?=${PROW_JOB_ID}
-GO_TEST ?= go test
+
+SHELL := /bin/bash
+
+MAKEFILE_PATH := $(abspath $(lastword $(MAKEFILE_LIST)))
+ROOT_DIR := $(dir $(MAKEFILE_PATH))
+
 ARTIFACTS_BUCKET?=my-s3-bucket
 GIT_VERSION?=$(shell git describe --tag)
 GIT_TAG?=$(shell git describe --tag | cut -d'-' -f1)
 GOLANG_VERSION?="1.16"
+GO ?= source $(ROOT_DIR)/scripts/common.sh && build::common::use_go_version $(GOLANG_VERSION) && go
```

I've been marinating on this line since yesterday... I kind of dig it, but its a bit "hacky" looking (reminds me of every bit of code I've ever written :)).  I think maybe a slightly better version would be:

```suggestion
GO ?= source $(ROOT_DIR)/scripts/common.sh && $(build::common::get_go_path $(GOLANG_VERSION))/go
```

Double check it still works, but that will avoid manipulating the path which `use_go_version` does.  I think we should be good since we are explicitly calling go here when using this approach.  In other repos where we use this same technique we need to change the PATH since we sometimes calling upstream makefiles/scripts.

---

I like it. Let me give it a test

---

I applied your change with slight mod as seen below and all is well.

`GO ?= $(shell source ./scripts/common.sh && build::common::get_go_path $(GOLANG_VERSION))/go`